sklearn_numba_dpex/testing/config.py

```diff
@@ -31,14 +31,15 @@ def override_attr_context(obj, **attrs):
     overriden. The initial values are restored when exiting the context.
 
     Trying to override attributes that don't exist will result in an AttributeError"""
-
-    attrs_before = dict()
-    for attr_name, attr_value in attrs.items():
-        # raise AttributeError if obj does not have the attribute attr_name
-        attrs_before[attr_name] = getattr(obj, attr_name)
-        setattr(obj, attr_name, attr_value)
-
-    yield
-
-    for attr_name, attr_value in attrs_before.items():
-        setattr(obj, attr_name, attr_value)
+    try:
+        attrs_before = dict()
+        for attr_name, attr_value in attrs.items():
+            # raise AttributeError if obj does not have the attribute attr_name
+            attrs_before[attr_name] = getattr(obj, attr_name)
+            setattr(obj, attr_name, attr_value)
+
+        yield
+
+    finally:
+        for attr_name, attr_value in attrs_before.items():
+            setattr(obj, attr_name, attr_value)
```

Comment:
How about this?
```suggestion
      attrs_before = dict()
      for attr_name, attr_value in attrs.items():
          attribute = getattr(obj, attr_name, None)
          if attribute is not None:
              # Only replace the value of the `attr_name` does exist.
              attrs_before[attr_name] = attribute
              setattr(obj, attr_name, attr_value)

      yield

      for attr_name, attr_value in attrs_before.items():
          setattr(obj, attr_name, attr_value)
```

Reply:
The `try/finally` is not only about the `AttributeError` that was documented, but about any error that can be raised after `yield` and before exiting the context. I thought initially that before exiting the code after `yield` is always executed, in the same way it's true for  `__exit__` when expliciting context managers, but in case of such errors it seems not to be true, and caused failing tests (in the sklearn pipeline). Adding the `try/finally` fixes that.

About only replacing the value if `attr_name` does exist, i'd prefer not, raising an `AttributeError` is a safeguard against typos, or unseen changes in input objects, when using this context manager. Its only intended uses are for replacing attributes that exist and I like it to fail otherwise.